pkg/convert/tf.go

```diff
@@ -1722,6 +1737,230 @@ func bodyContent(body hcl.Body) *hcl.BodyContent {
 	return content
 }
 
+// A specialized version of convertBody that is specific to converting the body of helm_release resources
+func convertHelmReleaseResource(state *convertState, scopes *scopes, fullyQualifiedPath string, body hcl.Body) bodyAttrsTokens {
+	contract.Assertf(fullyQualifiedPath != "", "fullyQualifiedPath should not be empty")
+
+	content := bodyContent(body)
+	newAttributes := make(bodyAttrsTokens, 0)
+	valueAttributes := make([]hclwrite.ObjectAttrTokens, 0)
+	firstValueLine := 0
+
+	for _, block := range content.Blocks {
+		if block.Type == "timeouts" {
+			// Timeouts are a special resource option block, we can't currently convert that PCL so just skip
+			continue
+		}
+
+		blockPath := appendPath(fullyQualifiedPath, block.Type)
+		if block.Type == "dynamic" {
+			// For dynamic blocks the path is the first label, not "dynamic"
+			blockPath = appendPath(fullyQualifiedPath, block.Labels[0])
+		}
+		// If this is a list so add [] to the path
+		isList := !scopes.maxItemsOne(blockPath)
+		name := scopes.pulumiName(blockPath)
+		if isList {
+			blockPath = appendPathArray(blockPath)
+		}
+
+		if block.Type == "dynamic" {
```

Comment:
This all seems like generic parsing logic, is it really resource-specific?

Reply:
This is not resource specific, `dynamic` blocks are used to create list comprehensions for resource fields. That said, for this resource, I don't see this being used for the available fields. I can simplify and remove this logic from here.